### PR TITLE
Fix crash in stateless_random_binomial when shape type is int64

### LINKED ISSUE #59160: Check failure when running gen_stateless_random_ops.stateless_random_binomial
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Mobile device

_No response_

### Python version

3.9

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current Behaviour?

```shell
I get check failure with the following input combination.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_stateless_random_ops
try:
  shape_tensor = tf.random.uniform([1], minval=-256, maxval=257, dtype=tf.int64)
  shape = tf.identity(shape_tensor)
  seed_0 = 12
  seed_1 = 34
  seed = [seed_0,seed_1,]
  counts_tensor = tf.random.uniform([], dtype=tf.float32)
  counts = tf.identity(counts_tensor)
  probs_tensor = tf.random.uniform([], dtype=tf.float32)
  probs = tf.identity(probs_tensor)
  dtype = tf.float16
  out = gen_stateless_random_ops.stateless_random_binomial(shape=shape,seed=seed,counts=counts,probs=probs,dtype=dtype,)
except Exception as e:
  print("Error:"+str(e))
```

```
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import stateless_random_ops
try:
  shape_0 = 125091515651
  shape = [shape_0,]
  seed_0 = 12
  seed_1 = 34
  seed = [seed_0,seed_1,]
  counts = 10.0
  probs = 0.4
  output_dtype = tf.float16
  out = stateless_random_ops.stateless_random_binomial(shape=shape,seed=seed,counts=counts,probs=probs,output_dtype=output_dtype,)
except Exception as e:
  print("Error:"+str(e))
```

### Relevant log output

```shell
Error:{{function_node __wrapped__StatelessRandomBinomial_device_/job:localhost/replica:0/task:0/device:CPU:0}} Dimension -144 must be >= 0 [Op:StatelessRandomBinomial]
2023-01-07 13:26:54.833378: F tensorflow/core/framework/tensor.cc:719] Check failed: dtype() == expected_dtype (9 vs. 3) int32 expected, got int64
Aborted

```
```
</details>

### PR DESCRIPTION
**Pull request by yongtang**
This PR tries to address the issue raised in 59160 where stateless_random_binomial will crash when shape type is int64.

This PR fixes #59160.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/kernel_tests/random/random_binomial_test.py [hunk 1]
```diff
@@ -207,6 +208,24 @@ def moment(self, i):
         )
         self.assertAllLess(z_scores, z_limit)
 
+  def testStatelessDtypes(self):
+    # Test case for GitHub issue 59160
```
Comment (vufg) [inline tensorflow/python/kernel_tests/random/random_binomial_test.py:212]: nit: remove this comment
Ideally the test name should explain itself, therefore there is no need to include github issue id here.

File: tensorflow/python/kernel_tests/random/random_binomial_test.py [hunk 2]
```diff
@@ -207,6 +208,24 @@ def moment(self, i):
...
+    shape_0 = constant_op.constant(1, dtype=dtypes.int64) #125091515651
```
Comment (vufg) [inline tensorflow/python/kernel_tests/random/random_binomial_test.py:213]: what does "#125091515651" mean here?

Comment (vufg) [reply tensorflow/python/kernel_tests/random/random_binomial_test.py:213]: do we need to test other types here? @cantonios

File: tensorflow/python/kernel_tests/random/random_binomial_test.py (hunk 1 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/random/random_binomial_test.py]: Thanks @vufg. The comment has been removed.

File: tensorflow/python/kernel_tests/random/random_binomial_test.py (hunk 2 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/random/random_binomial_test.py]: @vufg The `125091515651` was the initial "large" number I tried to test. Then I noticed that any number with int64 type will trigger the crash (before the fix). Therefore I changed the value from 125091515651 to 1 to make test simple.

**Comment by yongtang** [comment]
The PR has been updated now. Please take a look. /cc @vufg @cantonios

**Review by cantonios — APPROVED** [review]

**Pull request merged**